bin/casher

```diff
@@ -130,18 +130,20 @@ class Casher
       return
     end
 
-    msg "changes detected, packing new archive"
-
     @push_tar  = File.expand_path('push.tgz', @casher_dir) if path_ext(url) == 'tgz'
 
+    msg "changes detected, packing new archive"
+
     tar(:c, @push_tar, *cached_directories) do
       @counter += 1
       puts "." if @counter % 5 == 0
       sleep 1
     end
 
-    msg "uploading archive"
-    unless system "curl -T %p %p -f -v >#{@casher_dir}/push.log 2>#{@casher_dir}/push.err.log" % [@push_tar, url]
+    msg "uploading #{%r(([^/]+?\/[^/]+?)(\?.*)?\?).match(url)[1]}"
```

Comment:
Getting better! Let's extract `%r(([^/]+?\/[^/]+?)(\?.*)?\?).match(url)[1]` to a method, so we don't have to repeat it here.